activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -100,6 +100,8 @@ def create_binds_for_hash(attributes)
             # For polymorphic relationships, find the foreign key and type:
             # PriceEstimate.where(estimate_of: treasure)
             result[column_name] = AssociationQueryHandler.value_for(table, column_name, value)
+          when value.is_a?(Array) && !table.type(column_name).respond_to?(:subtype)
```

Comment:
We need to produce binds in `create_binds_for_hash` at least for now because handlers doesn't support to produce binds.

For example #26202, #26098, and 6efb394. These changes are not necessary if we don't need boundable queries in these case. But if we need boundable queries, we need to preprocess it before passing handlers.

In this PR. If a value includes `Relation`, we always need to extract `value.bound_attributes` because `RelationHandler` uses `value.arel` (e1533d2).